src/Nethermind/Nethermind.Evm/Tracing/GethStyle/Custom/Native/Tracers/Native4ByteTracer.cs

```diff
@@ -0,0 +1,92 @@
+// SPDX-FileCopyrightText: 2024 Demerzel Solutions Limited
+// SPDX-License-Identifier: LGPL-3.0-only
+
+using System;
+using System.Collections.Generic;
+using Nethermind.Core;
+using Nethermind.Core.Extensions;
+using Nethermind.Int256;
+namespace Nethermind.Evm.Tracing.GethStyle.Custom.Native.Tracers;
+
+// 4byteTracer searches for 4byte-identifiers, and collects them for post-processing.
+// It collects the methods identifiers along with the size of the supplied data, so
+// a reversed signature can be matched against the size of the data.
+//
+// Example:
+//   > debug.traceTransaction( "0x214e597e35da083692f5386141e69f47e973b2c56e7a8073b1ea08fd7571e9de", {tracer: "4byteTracer"})
+//   {
+//     0x27dc297e-128: 1,
+//     0x38cc4831-0: 2,
+//     0x524f3889-96: 1,
+//     0xadf59f99-288: 1,
+//     0xc281d19e-0: 1
+//   }
+public sealed class Native4ByteTracer : GethLikeNativeTxTracer
+{
+    public const string _4byteTracer = "4byteTracer";
+
+    private readonly Dictionary<string, int> _4ByteIds = new();
+    private Instruction _op;
+
+    public Native4ByteTracer(
+        GethTraceOptions options) : base(options)
+    {
+        IsTracingActions = true;
+        IsTracingMemory = true;
+        IsTracingStack = true;
+    }
+
+    protected override GethLikeTxTrace CreateTrace() => new();
+
+    public override GethLikeTxTrace BuildResult()
+    {
+        GethLikeTxTrace result = base.BuildResult();
+
+        result.CustomTracerResult = new GethLikeCustomTrace() { Value = _4ByteIds };
+        return result;
+    }
+
+    public override void ReportAction(long gas, UInt256 value, Address from, Address to, ReadOnlyMemory<byte> input, ExecutionType callType, bool isPrecompileCall = false)
+    {
+        base.ReportAction(gas, value, from, to, input, callType, isPrecompileCall);
+
+        if (_depth == 0)
+        {
+            CaptureStart(input);
+        }
+        else
+        {
+            CaptureEnter(_op, input, to, isPrecompileCall);
+        }
```

Comment:
Maybe move this logic to parent class? Depends if useful for other tracers.

Reply:
I'm going to leave this in the native 4byteTracer in this PR but may refactor it into the parent class in the PR for adding the prestate or call tracer if it's needed there too